File: C#/Source/GitHubUpdater/Classes/Assets.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TheE7Player
{
    /// <summary>
    /// Class which holds the file information from Releases
    /// </summary>
    public class Assets
    {
        /// <summary>
        /// Enum to display file size to a specific size (Used to convert)
        /// </summary>
        public enum DisplaySize { Kilobytes = 2, Megabytes = 3, Gigabytes = 4};

        #region Variables/Fields
        private string itemname, itemdownloadurl;
        private double itemsize;

        public Assets(string Name, string URL, double Size) { this.itemName = Name; this.itemDownloadUrl = URL; this.itemSize = Size; }

        /// <summary>
        /// Holds the items name
        /// </summary>
        public string itemName
        {
            get { return itemname; }
            private set { itemname = value; }
        }

        /// <summary>
        /// Holds the items downloadable url link
        /// </summary>
        public string itemDownloadUrl
        {
            get { return itemdownloadurl; }
            private set { itemdownloadurl = value; }
        }

        /// <summary>
        /// Holds the item size
        /// </summary>
        public double itemSize
        {
            get { return itemsize; }
            private set { itemsize = value; }
        }
        #endregion

        #region Functions
        public string getItemSize(DisplaySize SizeType)
        {
            //Fetch amount of times to divide by from the SizeType declared
            int IterationAmount = (int)SizeType; //Fetch the enum value by casting it into an int (To receive it's assigned number)
            double initialSize = itemSize;

            for (int i = 1; i < IterationAmount; i++)
                initialSize /= 1024;

            string SizeExtension = (SizeType == DisplaySize.Kilobytes) ? "KB" : (SizeType == DisplaySize.Megabytes) ? "MB" : "GB";

            //Example return: 92.8MB
            return $"{initialSize.ToString("0.0")}{SizeExtension}";
        }
        #endregion
    }
}

File: Java/Source/TheE7Player/Assets.java
package TheE7Player;

public class Assets
{
    //[NOTE] API displays the file size in bytes!
    //[NOTE] Divide by 1024 twice for MB and triple for GB

    /*
        Example: Find 97318010 byte into MB or GB
        1. 97318010 (bytes) / 1024 = 95,037.119... (Kilobytes)
        2. 95,037.119... (Kilobytes) / 1024 = (92.8)... (Megabytes) <--
        3. 92.8... (Megabytes) / 1024 = (0.09)0... (Gigabytes) <--
     */

    public enum DisplaySize
    {
        //https://stackoverflow.com/a/1067371 <- Java enum with numbers assigned

        Kilobytes(2),
        Megabytes(3),
        Gigabytes(4);

        private final int id;
        DisplaySize(int id) { this.id = id; }
        public int getValue() { return id; } // <- Use this to retrieve amount of times to divide by 1024
    }

    private String itemName; //<- Holds items name
    private String itemDownloadUrl; //<- Holds the items 'individual name'
    private double itemSize; //<- Holds the items size

    //Constructor to assign the private fields
    public Assets(String Name, String URL, double Size)
    {
        this.itemName = Name; this.itemDownloadUrl = URL; this.itemSize = Size;
    }

    //Getters to fetch the information
    public String getItemName() {
        return itemName;
    }
    public String getItemURL() {
        return itemDownloadUrl;
    }
    public String getItemSize(DisplaySize SizeType)
    {
        //Fetch amount of times to divide by from the SizeType declared
        int IterationAmount = SizeType.getValue();
        double initialSize = itemSize;

        for (int i = 1; i < IterationAmount; i++)
            initialSize /= 1024;

        String SizeExtension = (SizeType == DisplaySize.Kilobytes) ? "KB" :  (SizeType == DisplaySize.Megabytes) ? "MB" : "GB";

        return String.format("%.1f%s", initialSize, SizeExtension);
    }
}
